crates/pessimistic-proof/src/local_state.rs

```diff
@@ -67,8 +67,12 @@ impl LocalNetworkState {
         }
 
         // TODO: benchmark if BTreeMap is the best choice in terms of SP1 cycles
-        let mut new_balances: BTreeMap<_, _> =
-            multi_batch_header.balances_proofs.iter().map(|(k, v)| (*k, v.0)).collect();
+        let mut new_balances = BTreeMap::new();
+        for (k, v) in &multi_batch_header.balances_proofs {
+            if new_balances.insert(*k, v.0).is_some() {
+                return Err(ProofError::DuplicateTokenBalanceProof);
```

Comment:
Could we add information about the duplication in order to log them and detect potential problem with a CDK?

Reply:
Done